src/presentation/controller/common/GeneralPartnerController.ts

```diff
@@ -418,6 +390,41 @@ abstract class GeneralPartnerController extends AbstractController {
     };
   }
 
+  private async conditionalPreviousUrl(
+    ids: { transactionId: string; submissionId: string; generalPartnerId: string },
+    pageRouting: PageRouting,
+    request: Request,
+    tokens
+  ) {
+    const pageType = this.getJourneyPageTypes(request.url);
+
+    if (
+      pageRouting.pageType === pageType.addGeneralPartnerLegalEntity ||
+      pageRouting.pageType === pageType.addGeneralPartnerPerson
+    ) {
+      const result = await this.generalPartnerService.getGeneralPartners(tokens, ids.transactionId);
+
+      if (result.generalPartners.length > 0) {
+        pageRouting.previousUrl = super.insertIdsInUrl(pageRouting.data?.customPreviousUrl, ids, request.url);
+      }
+    }
+  }
+
+  private async conditionalNextUrl(
+    tokens: { access_token: string; refresh_token: string },
+    ids: { transactionId: string; submissionId: string },
+    pageRouting: PageRouting,
+    urls: {
+      reviewLimitedPartnersUrl: string;
+    }
+  ) {
+    const result = await this.limitedPartnerService.getLimitedPartners(tokens, ids.transactionId);
```

Comment:
is it ok for this function to get limited partners if this is the general partners controller